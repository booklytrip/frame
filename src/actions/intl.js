/**
 * @flow
 */

import type { Locale } from '../types';

/**
 * Change current locale to specified
 *
 * @param {String} locale - A new locale to set
 */
export const updateLocale = (locale: Locale) => ({
    type: 'INTL.UPDATE_LOCALE',
    locale,
});
