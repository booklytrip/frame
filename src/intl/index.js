import { addLocaleData } from 'react-intl';
import { reduce, mapValues } from 'lodash';

import en from 'react-intl/locale-data/en';
import lv from 'react-intl/locale-data/lv';
import ru from 'react-intl/locale-data/ru';

import * as defaultLocales from './locales';
import { locales as uikitLocales } from 'uikit';

addLocaleData([
    ...en,
    ...lv,
    ...ru,
]);

/**
 * Combine all messages from list of provides locales
 * and return single object
 */
const combineLocales = (locales) => (
    reduce(locales, (result, locale) => (
        mapValues(locale, (messages, language) => (
            { ...result[language], ...messages }
        ))
    ), {})
);

const locales = combineLocales([defaultLocales, uikitLocales]);

export { locales };
