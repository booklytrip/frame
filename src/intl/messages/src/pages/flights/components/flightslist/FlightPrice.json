[
  {
    "id": "flights.flightslist.FlightPrice.total",
    "defaultMessage": "Total price: â‚¬ {totalPrice}"
  }
]