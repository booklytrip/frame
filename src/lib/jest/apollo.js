import React from 'react';

import { ApolloProvider as Provider } from 'react-apollo';
import ApolloClient from 'apollo-client';
const client = new ApolloClient();

export const ApolloProvider = ({ children }) => (
    <Provider client={ client }>
        { children }
    </Provider>
);
