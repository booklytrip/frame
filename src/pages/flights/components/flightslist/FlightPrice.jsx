/**
 * @flow
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import { withProps } from 'recompose';
import { split } from 'lodash';

type FlightPriceProps = {
    price: Array<number>,
    totalPrice: ?number,
    currency: string,
};

const RootSegment = styled.div`margin-top: 10px;`;

const PriceAmount = styled.div`
    font-size: 36px;
    font-weight: 600;
    display: inline-block;
    color: #e92133;
`;

const PriceDecimals = styled.sup`
    font-size: 16px;
    font-weight: 400;
    top: -16px;
`;

const Currency = styled.span`
    font-size: 18px;
    padding-right: 5px;
    font-weight: 400;
`;

const TotalPrice = styled.p`
    color: #636262;
    font-size: 13px;
    margin-top: 3px;
`;

const FlightPrice = ({ price, totalPrice, currency }: FlightPriceProps) => (
    <RootSegment>
        <PriceAmount>
            <Currency>€</Currency>
            {price[0]}
            <PriceDecimals>{price[1]}</PriceDecimals>
        </PriceAmount>

        {totalPrice && (
            <TotalPrice>
                <FormattedMessage
                    id="flights.flightslist.FlightPrice.total"
                    defaultMessage={`Total price: € {totalPrice}`}
                    values={{
                        totalPrice: parseFloat(totalPrice).toFixed(2),
                    }}
                />
            </TotalPrice>
        )}
    </RootSegment>
);

const props = withProps(ownProps => ({
    price: split(
        parseFloat(ownProps.price)
            .toFixed(2)
            .toString(),
        '.',
    ),
}));

export default props(FlightPrice);
