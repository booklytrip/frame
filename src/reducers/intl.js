/**
 * @flow
 */

import { Map } from 'immutable';
import { saveState, loadState } from '../lib/sessionStorage';

const initialState = new Map(loadState('intl') || {});

export default function(state: Object = initialState, action: Object) {
    switch (action.type) {
        // Replace current locale with specified
        case 'INTL.UPDATE_LOCALE': {
            const newState = state.set('locale', action.locale);
            saveState('intl', newState);
            return newState;
        }
        default:
            return state;
    }
}
