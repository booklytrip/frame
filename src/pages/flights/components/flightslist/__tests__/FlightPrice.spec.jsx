import React from 'react';
import { mountWithIntl, createWithIntl } from '../../../../../lib/jest/intl';
import FlightPrice from '../FlightPrice';

describe('FlightPrice', () => {
    it('renders without crashing', () => {
        mountWithIntl(
            <FlightPrice
                price={{
                    amount: 100,
                    currency: 'EUR',
                }}
            />,
        );
    });

    describe('Snapshots', () => {
        it('renders flight price', () => {
            const rendered = createWithIntl(
                <FlightPrice
                    price={{
                        amount: 100,
                        currency: 'EUR',
                    }}
                />,
            );
            expect(rendered.toJSON()).toMatchSnapshot();
        });
    });
});
