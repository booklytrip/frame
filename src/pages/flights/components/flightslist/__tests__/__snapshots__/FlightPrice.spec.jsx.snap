// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightPrice Snapshots renders flight price 1`] = `
<div
  className="ifExBw"
>
  <div
    className="foaqzg"
  >
    <span
      className="IrIgN"
    >
      â‚¬
    </span>
    100
    <sup
      className="hhveXe"
    >
      00
    </sup>
  </div>
  <p
    className="cpOoys"
  >
    <span>
      Per person
    </span>
  </p>
</div>
`;
